packages/react-native/ReactAndroid/src/main/jni/CMakeLists.txt

```diff
@@ -249,6 +249,9 @@ target_link_libraries(reactnative
             yogacore
 )
 
+# Enable aggressive compiler optimizations and link-time-optimizations
+target_compile_options(reactnative PRIVATE -O3 -flto)
```

Comment:
So in theory this should not be needed, because the Android NDK toolchain already should be setting those flags for us.

Specifically the release versions of React Native are already built with:

```
LANGUAGE_COMPILE_FLAGS = -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D__BIONIC_NO_PAGE_SIZE_MACRO -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security   -O3 -DNDEBUG  -flto=thin
```